ckanext/advancedauth/auth.py

```diff
@@ -86,7 +87,11 @@ def only_approved_users(context, data_dict=None):
     try:
         user_id = context.get("auth_user_obj").id
     except AttributeError:
-        user_id = context.get("user_obj").id
+        try:
+            user_id = context.get("user_obj").id
+        except AttributeError:
+            user = _get_user(context.get("user"))
```

Comment:
A third attempt to get `user`-- when is this necessary?

Reply:
Hmm I can't remember the details exactly but I believe after adding sysadmins, this portion generated exceptions when trying to access `auth_user_obj` and `user_obj` so I've added the function to get the user object directly from the db for this special case. Though I don't know why this happens

Reply:
I'll double check since we're already fetching the user earlier in the code, so this doesn't seem necessary anymore.

Reply:
Reverted this third attempt code - no longer needed